src/assets/scripts/client/aircraft/AircraftTypeDefinitionModel.js

```diff
@@ -203,4 +203,24 @@ export default class AircraftTypeDefinitionModel extends BaseModel {
 
         return aircraftIcao.toUpperCase();
     }
+
+    /**
+     * @for AircraftTypeDefinitionModel
+     * @method assignedAttainableAltitude
+     * @param altitude
+     * @return {Boolean}
+     */
+    assignedAttainableAltitude(altitude) {
```

Comment:
Still not loving the name here... since it's a method of the aircraft definition, we're basically asking:

`Boeing 737 with winglets, are you assigned an attainable altitude?`

This sounds like it wouldn't take an argument. And it needs to take one, because the type definition won't have access to the MCP of a particular instance. Therefore, I think we'd be better with:

`Boeing 737 with winglets, are you able to maintain 330 knots?`

This would result in a name of `AircraftTypeDefinitionModel.isAbleToMaintainAltitude(altitude)`. I think that form for the altitude and speed methods will make much more sense. It's a bit clunkier, but overall more clear I think.